src/app/(frontend)/embeds/courses/page.tsx

```diff
@@ -0,0 +1,111 @@
+import { getCourses } from '@/actions/getCourses'
+import { getCoursesStates } from '@/actions/getCoursesStates'
+import { getProviders } from '@/actions/getProviders'
+import { CoursesList } from '@/components/CoursesList'
+import { AffinityGroupsFilter } from '@/components/filters/AffinityGroupsFilter'
+import { ModesOfTravelFilter } from '@/components/filters/ModesOfTravelFilter'
+import { ProvidersFilter } from '@/components/filters/ProvidersFilter'
+import { StatesFilter } from '@/components/filters/StatesFilter'
+import { FiltersTotalProvider } from '@/contexts/FiltersTotalContext'
+import { createLoader, parseAsBoolean, parseAsString, SearchParams } from 'nuqs/server'
+import { CoursesDateFilter } from './courses-date-filter'
+import { CoursesMobileFilters } from './courses-mobile-filters'
+import { CoursesTypeFilter } from './courses-type-filter'
+
+const coursesSearchParams = {
+  backgroundColor: parseAsString,
+  title: parseAsString,
+  showFilters: parseAsBoolean.withDefault(false),
+  types: parseAsString,
+  providers: parseAsString,
+  states: parseAsString,
+  affinityGroups: parseAsString,
+  modesOfTravel: parseAsString,
+  startDate: parseAsString,
+  endDate: parseAsString,
+}
+const loadSearchParams = createLoader(coursesSearchParams)
+
+export default async function CoursesEmbedPage({
+  searchParams,
+}: {
+  searchParams: Promise<SearchParams>
+}) {
+  const {
+    backgroundColor,
+    title,
+    showFilters,
+    types,
+    providers,
+    states,
+    affinityGroups,
+    modesOfTravel,
+    startDate,
+    endDate,
+  } = await loadSearchParams(searchParams)
+
+  const filters = {
+    types,
+    providers,
+    states,
+    affinityGroups,
+    modesOfTravel,
+    startDate,
+    endDate,
+  }
+
+  const { courses, hasMore, total, error } = await getCourses(filters)
+
+  const { providers: providersList } = showFilters ? await getProviders() : { providers: [] }
+  const { states: statesList } = showFilters ? await getCoursesStates() : { states: [] }
+
+  const hasActiveFilters = Boolean(
+    types || providers || states || affinityGroups || modesOfTravel || startDate || endDate,
+  )
+
+  return (
+    <FiltersTotalProvider initialTotal={total}>
+      <div
+        style={
+          backgroundColor && typeof backgroundColor === 'string' ? { backgroundColor } : undefined
+        }
```

Comment:
Can this be cleaned up?

```suggestion
        style={backgroundColor ? { backgroundColor } : undefined}
```

Reply:
Any reason we are using `style` here or can we use tailwind classes?

Reply:
Yep, can remove the typeof check now. This was leftover from before I was using `nuqs` to make the query params type safe - so now we don't need this type check. 

`style` is necessary here as opposed to tailwind classes because a user can pass any color - really any string - to the backgroundColor query param and you can't use arbitrary values that are dynamic: https://v3.tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
Make sense. I forgot the context of a user was passing in the params via string.